executor/executor_test.go

```diff
@@ -6809,6 +6809,87 @@ func (s *testSerialSuite) TestCoprocessorOOMAction(c *C) {
 	}
 }
 
+func (s *testSuite) TestInlineProjectionForSortAndTopN(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("create table t1 (a int(11) not null,b int(11) not null,c int(11) not null, PRIMARY KEY (a))")
+	tk.MustExec("insert into t1 values (3, 1, 2),(4, 2, 3),(5, 3, 4)")
+	tk.MustQuery("select * from t1 order by a").Check(testkit.Rows("3 1 2", "4 2 3", "5 3 4"))
+	tk.MustQuery("select * from t1 order by a desc").Check(testkit.Rows("5 3 4", "4 2 3", "3 1 2"))
+	tk.MustQuery("select c,b from t1 order by a").Check(testkit.Rows("2 1", "3 2", "4 3"))
+	tk.MustQuery("select c,b from t1 order by a desc").Check(testkit.Rows("4 3", "3 2", "2 1"))
+	tk.MustQuery("select a from t1 order by b, c").Check(testkit.Rows("3", "4", "5"))
+	tk.MustQuery("select a from t1 order by b desc, c").Check(testkit.Rows("5", "4", "3"))
+	tk.MustQuery("select b from t1 group by a order by a").Check(testkit.Rows("1", "2", "3"))
+	tk.MustQuery("select sum(c) from t1 order by a").Check(testkit.Rows("9"))
+	tk.MustQuery("select min(c) from t1 order by a limit 1").Check(testkit.Rows("2"))
+	tk.MustQuery("select max(c) from t1 order by a limit 1").Check(testkit.Rows("4"))
+	tk.MustQuery("select c from t1 order by a limit 1").Check(testkit.Rows("2"))
+	tk.MustQuery("select c from t1 order by a desc limit 1").Check(testkit.Rows("4"))
+	tk.MustQuery("select ta.a, ta.b from t1 ta left join t1 tb on ta.a = tb.a order by tb.a").Check(
+		testkit.Rows("3 1", "4 2", "5 3"))
+	tk.MustQuery("select ta.c, ta.b from t1 ta left join t1 tb on ta.a = tb.a order by tb.a").Check(
+		testkit.Rows("2 1", "3 2", "4 3"))
+	tk.MustQuery("select b from t1 order by 1").Check(testkit.Rows("1", "2", "3"))
+	tk.MustQuery("select b from t1 order by c+1").Check(testkit.Rows("1", "2", "3"))
+	tk.MustQuery("select ta.a from t1 ta, t1 tb where ta.b = tb.a order by 3*ta.a").Check(
+		testkit.Rows("5"))
+}
+
+func (s *testSerialSuite) TestIssue20454(c *C) {
```

Comment:
There seems to be a wrong merge. This test case is removed in #21168

Reply:
wow, I did merge a conflict. I will fix it.